extension/observer/hostobserver/extension.go

```diff
@@ -35,6 +35,11 @@ type hostObserver struct {
 type endpointsLister struct {
 	logger       *zap.Logger
 	observerName string
+
+	// For testing
+	getConnections        func() ([]net.ConnectionStat, error)
+	newProcess            func(pid int32) (*process.Process, error)
```

Comment:
Since you're now in charge of naming, maybe something more informative like `getProcess`?

Reply:
Done.